src/Components/NotFoundErrorPage.tsx

```diff
@@ -43,3 +60,27 @@ export const NotFoundErrorPage = connect(function NotFoundErrorPage() {
     </>
   )
 })
+
+const NoRootError = connect(
+  function NoRootError() {
+    Obj.root() // read root again, to trigger "loading" state
+
+    useEffect(() => {
+      load(() => Obj.root()).then((root) => {
+        if (root === null && !isUserLoggedIn()) ensureUserIsLoggedIn()
+      })
+    }, [])
+
+    return <div>Page not found.</div>
+  },
+  { loading: () => null },
+)
+
+const PortalSiteNotFound = connect(
+  function PortalSiteNotFound() {
+    if (portalSiteAvailable()) return null
+
+    return <div>Site not found.</div>
```

Comment:
* [ ] FYI I don't see how a user would profit from "Site not found" vs. "Page not found". The distinction may be more confusing than helpful. How about showing `<NotFound />` here, too? 
* [ ] This implementation is also missing the auto-login feature for login-only websites. `<NotFound />` would solve it.